datadog/resource_datadog_cloud_configuration_rule.go

```diff
@@ -0,0 +1,296 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/validators"
+)
+
+const nameField = "name"
+const messageField = "message"
+const enabledField = "enabled"
+const policyField = "policy"
+const resourceTypeField = "resource_type"
+const relatedResourceTypesField = "related_resource_types"
+const severityField = "severity"
+const notificationsField = "notifications"
+const groupByField = "group_by"
+const tagsField = "tags"
+
+func resourceDatadogCloudConfigurationRule() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog Cloud Configuration Rule resource.",
+		CreateContext: cloudConfigurationRuleCreateContext,
+		ReadContext:   cloudConfigurationRuleReadContext,
+		UpdateContext: cloudConfigurationRuleUpdateContext,
+		DeleteContext: resourceDatadogSecurityMonitoringRuleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: cloudConfigurationRuleSchema(),
+	}
+}
+
+func cloudConfigurationRuleSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		nameField: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The name of the cloud configuration rule.",
+		},
+		messageField: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The message associated to the rule that will be shown in findings and signals.",
+		},
+		enabledField: {
+			Type:        schema.TypeBool,
+			Required:    true,
+			Description: "Whether the cloud configuration rule is enabled.",
+		},
+		policyField: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Policy written in Rego format.",
+		},
+		resourceTypeField: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Main resource type to be checked by the rule.",
+		},
+		relatedResourceTypesField: {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "Related resource types to be checked by the rule. Defaults to empty list.",
+			MinItems:    0,
+			MaxItems:    9,
```

Comment:
Did we decided to go with max 9 in the end? I thought it was 5 last time I followed the conversation.

Reply:
The code in the backend limits to 10 resource types (main resource type + related resource types): https://github.com/DataDog/logs-backend/blob/prod/domains/cloud-security-platform/apps/security-monitoring-api/src/main/java/com/dd/logs/api/rule_engine/validators/compliance/CloudConfigurationRuleValidator.java#L209